flutter_mobx/lib/flutter_mobx.dart

```diff
@@ -0,0 +1,57 @@
+library flutter_mobx;
+
+import 'package:flutter/widgets.dart';
+import 'package:mobx/mobx.dart';
+
+typedef ObserverBuilder = Widget Function(BuildContext);
+
+class Observer extends StatefulWidget {
+  const Observer({@required this.builder, Key key, this.context})
+      : assert(builder != null),
+        super(key: key);
+
+  final ReactiveContext context;
+  final ObserverBuilder builder;
+
+  @visibleForTesting
+  DerivationTracker createDerivationTracker(Function() onInvalidate) =>

```

Comment:
@katis Wondering if we even need a **DerivationTracker**. Why not use the **Reaction** directly, now that we can expose the `{start, end}Tracking` as separate calls? 🤔

Reply:
Frankly, I really don't like using the start/endTracking methods anywhere. The `track(callback)` interface is a superior one from a safety point of view. Consider a function like this:

```dart
foo() async {
  tracker.startTracking();
  await doSomethingAsync();
  tracker.endTracking();
}
```

Here the tracking starts, `doSomethingAsync` is called, returning a Future and the method is suspended. Then the Dart runtime schedules something else to execute while waiting the Future to resolve, making mobx track all sorts of random things. You can't create a mess like this with the callback interface. This is why the only place to use it should be the `useObserver` hook...

I think I'll figure a way to expose `Reaction.track` properly and use that instead here, for the above reasons, but I'll do it in a separate PR.